lib/core/controlplane.rb

```diff
@@ -149,6 +149,26 @@ def fetch_workload_deployments(workload)
     api.workload_deployments(workload: workload, gvc: gvc, org: org)
   end
 
+  def deployment_version_ready?(version, next_version, ready:)
+    return false unless version["workload"] == next_version
+
+    version["containers"]&.all? do |_, container|
+      ready_status = container.dig("resources", "replicas") == container.dig("resources", "replicasReady")
+      ready ? ready_status : !ready_status
+    end
+  end
+
+  def wait_for_workload_deployments(workload, ready:)
+    deployments = fetch_workload_deployments(workload)["items"]
+    deployments.all? do |deployment|
+      next_version = deployment.dig("status", "expectedDeploymentVersion")
+
+      deployment.dig("status", "versions")&.all? do |version|
+        deployment_version_ready?(version, next_version, ready: ready)
+      end
```

Comment:
I don't understand how this waits. Should there be some `sleep` and a loop?

Reply:
It doesn't. The method that waits is `step`, which calls this method:

![step](https://github.com/shakacode/heroku-to-control-plane/assets/25509361/f775e499-d7d0-4e57-9129-e637a221da8f)

Then it keeps sleeping and looping for as long as `wait_for_workload_deployments` returns `false`.

I think this was a leftover of the initial implementation I made and I forgot to rename it.

I've renamed it to `workload_deployments_ready?` in c02a5ef829a042f7fe6bc29f6e432d76975323c6, and also renamed the param to `expected_status`, which makes more sense.